### ISSUE TITLE
Concurrency / time-slicing by default

### ISSUE THREAD

**Issue by rybon** [facebook/react#21662]
Will concurrency / time-slicing ever become the default behavior? Or will it always require a startTransition block to opt-in?
If it will become the default at some point, will that become part of the React 18 timeline or a later major release?

**Comment by gaearon** [comment]
For updates caused by [discrete](https://github.com/reactwg/react-18/discussions/46#discussioncomment-847382) user interactions like click/typing, it won’t ever be the default because those must execute sequentially and usually require immediate feedback.

For updates originating outside of such events — for example, from a fetch callback — it’s possible that we’ll make time slicing default in a future major release (not in 18 but seem future one), or that we would warn if you don’t use startTransition. This is still up in the air and very much depends on the state of the ecosystem compatibility with concurrency in the future. We definitely  would *not* change the behavior during the React 18.x release line because that would’ve been a breaking change. For now it’s a bit early to decide how future major versions should work.

**Comment by tlgimenes** [comment]
Hello. I'm testing React 18 features and I would like to test the time slicing feature. I've seen that [this PR](https://github.com/facebook/react/pull/21227) adds an option to createRoot for using time slicing. The problem I'm having is that this options seems to be behind a flag called `allowConcurrentByDefault` that is not enabled for `@experimental` or `@alpha` releases of React. Is it possible to enable this feature flag for `@experimental` releases?

Thanks!

**Comment by gaearon** [comment]
The way you opt into time slicing in React 18 is per-update. You do that by wrapping that update into `startTransition`.

Explanation: https://github.com/reactwg/react-18/discussions/41
Demo: https://github.com/facebook/react/tree/master/fixtures/concurrent/time-slicing

**Comment by rybon** [comment]
@gaearon makes sense. Thank you for the explanation.

Yeah, making a distinction between updates that require immediate feedback versus ones that don't and thus are eligible for time-slicing is something that requires an explicit API such as startTransition I suppose. Was wondering if it would also be generally beneficial on first render, but I guess the Suspense boundaries take care of that per subtree anyway?

Thanks for the awesome work React team!

**Comment by rybon** [comment]
Other question, does the implementation of the scheduler still make use of requestIdleCallback? I believe WebKit is the only holdout for that one.

**Comment by gaearon** [comment]
>Other question, does the implementation of the scheduler still make use of requestIdleCallback? 

No, it fired to late and we'd waste CPU time. It's really important for our use case that we utilize CPU to full extent rather than only after some idle period. So instead we rewrote to have our own loop that yields every 5ms.
